core/src/test/java/kafka/test/junit/ClusterTestExtensions.java

```diff
@@ -119,7 +139,27 @@ public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContex
         return generatedContexts.stream();
     }
 
+    @Override
+    public void beforeEach(ExtensionContext context) {
+        DetectThreadLeak detectThreadLeak = DetectThreadLeak.of(thread ->
+                SKIPPED_THREAD_PREFIX.stream().noneMatch(prefix -> thread.getName().startsWith(prefix)));
+        getStore(context).put(DETECT_THREAD_LEAK_KEY, detectThreadLeak);
+    }
+
+    @Override
+    public void afterEach(ExtensionContext context) throws InterruptedException {
+        DetectThreadLeak detectThreadLeak = getStore(context).remove(DETECT_THREAD_LEAK_KEY, DetectThreadLeak.class);
+        if (detectThreadLeak == null) {
+            return;
+        }
+        TestUtils.waitForCondition(() -> detectThreadLeak.newThreads().isEmpty(),
+                "Thread leak detected: " +
+                        detectThreadLeak.newThreads().stream().map(Thread::getName).collect(Collectors.joining(", ")));
+    }
 
+    private Store getStore(ExtensionContext context) {
+        return context.getStore(Namespace.create(getClass(), context.getUniqueId()));
```

Comment:
`getClass()` we don't need this as it is always the same for all test cases.

Reply:
Yes, using `getUniqueId` is enough. Thanks for the suggestion.